worlds/satisfactory/GameLogic.py

```diff
@@ -0,0 +1,991 @@
+
+liquids: set[str] = {
+    "Water",
+    "Liquid Biofuel",
+    "Crude Oil",
+    "Fuel",
+    "Heavy Oil Residue",
+    "Turbofuel",
+    "Alumina Solution",
+    "Sulfuric Acid",
+    "Nitrogen Gas",
+    "Nitric Acid",
+    "Dissolved Silica",
+    "Rocket Fuel",
+    "Ionized Fuel",
+    "Excited Photonic Matter",
+    "Dark Matter Residue"
+}
+
+radio_actives: set[str] = {
+    "Uranium",
+    "Encased Uranium Cell",
+    "Uranium Fuel Rod"
+    "Uranium Waste",
+    "Non-fissile Uranium",
+    "Plutonium Pellet",
+    "Encased Plutonium Cell",
+    "Plutonium Fuel Rod",
+    "Plutonium Waste",
+    "Ficsonium",
+    "Ficsonium Fuel Rod"
+}
+
+class Recipe():
+    """
+    Relationship between components and what is required to produce them (input ingredients, production building, etc.)
+    Not all recipes are Satisfactory FGRecipes - for example, Water has a Recipe, but it's not an FGRecipe
+    """
+    name: str
+    building: str
+    inputs: tuple[str, ...]
+    minimal_belt_speed: int
+    handcraftable: bool
+    implicitly_unlocked: bool
+    """No explicit location/item is needed to unlock this recipe, you have access as soon as dependencies are met (ex. Water, Leaves, tutorial starting items)"""
+    additional_outputs: tuple[str, ...]
+    minimal_tier: int
+
+    needs_pipes: bool
+    is_radio_active: bool
+
+    def __init__(self, name: str, building: Optional[str] = None, inputs: Optional[tuple[str, ...]] = None,
+            minimal_belt_speed: int = 1, handcraftable: bool = False, implicitly_unlocked: bool = False,
+            additional_outputs: Optional[tuple[str, ...]] = None, minimal_tier: Optional[int] = 1):
+        self.name = "Recipe: " + name
+        self.building = building
+        self.inputs = inputs
+        self.minimal_belt_speed = minimal_belt_speed
+        self.handcraftable = handcraftable
+        self.implicitly_unlocked = implicitly_unlocked
+        self.additional_outputs = additional_outputs
+        self.minimal_tier = minimal_tier
+
+        all_parts: list[str] = [name]
+        if inputs:
+            all_parts += inputs
+        if additional_outputs:
+            all_parts += additional_outputs
+
+        self.needs_pipes = not liquids.isdisjoint(all_parts)
+        self.is_radio_active = not radio_actives.isdisjoint(all_parts)
+
+class Building(Recipe):
+    power_requirement: Optional[PowerInfrastructureLevel]
+    can_produce: bool
+
+    def __init__(self, name: str, inputs: Optional[tuple[str, ...]] = None,
+            power_requirement: Optional[PowerInfrastructureLevel] = None, can_produce: bool = True,
+            implicitly_unlocked: bool = False):
+        super().__init__(name, None, inputs, handcraftable=True, implicitly_unlocked=implicitly_unlocked)
+        self.name = "Building: " + name
+        self.power_requirement = power_requirement
+        self.can_produce = can_produce
+        self.implicitly_unlocked = implicitly_unlocked
+
+
+class MamNode():

```

Comment:
```suggestion
class MamNode:
```
Don't need parentheses on a class definition with nothing in it